### PR TITLE
Code Coverage for NetCore

### PR DESCRIPTION
**Pull request by mayankbansal018**

### CODE REVIEW

File: src/Microsoft.TestPlatform.ObjectModel/DataCollector/DataCollectorSettings.cs [hunk 1]
```diff
@@ -210,6 +214,22 @@ internal static DataCollectorSettings FromXml(XmlReader reader)
                             XmlDocument doc = new XmlDocument();
                             XmlElement element = doc.CreateElement("Configuration");
                             element.InnerXml = reader.ReadInnerXml();
+
+                            // do not additional settings to InProcDataCollector, it might break LUT
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.ObjectModel/DataCollector/DataCollectorSettings.cs]: Nit : Update this comment.

File: src/vstest.console/Processors/CollectArgumentProcessor.cs [hunk 2]
```diff
@@ -155,13 +159,39 @@ internal static void EnableDataCollectorUsingFriendlyName(string argument, DataC
...
+
+            // todo: remove adding default configuration element once we pass full runsettings to datacollectors
+            if (dataCollectorSettings.Configuration == null)
             {
-                dataCollectorSettings.IsEnabled = true;
+                var doc = new XmlDocument();
+                using (
+                    var xmlReader = XmlReader.Create(
+                        new StringReader(DefaultConfigurationSettings),
+                        new XmlReaderSettings() { CloseInput = true, DtdProcessing = DtdProcessing.Prohibit }))
```
Comment (harshjain2) [inline src/vstest.console/Processors/CollectArgumentProcessor.cs]: Set  `XmlResolver=null`.

File: src/vstest.console/Resources/xlf/Resources.xlf [hunk 3]
```diff
@@ -643,8 +643,9 @@
       </trans-unit>
       <trans-unit id="CLIRunSettingsArgumentHelp">
         <source>RunSettings arguments:
-      Arguments to pass runsettings configurations through commandline. Arguments may be specified as name-value pair of the form [name]=[value] after "-- ". Note the space after --. Use a space to separate multiple [name]=[value].
-More info on RunSettings arguments support: https://aka.ms/vstest-runsettings-arguments</source>
+      Arguments to pass runsettings configurations through commandline. Arguments may be specified as name-value pair of the form [name]=[value] after "-- ". Note the space after --.
```
Comment (harshjain2) [inline src/vstest.console/Resources/xlf/Resources.xlf]: Include xlf files for other languages as well.

File: test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs [hunk 4]
```diff
@@ -1,6 +1,9 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
```
Comment (harshjain2) [inline test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs]: Move using statements below namespace.

File: test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs [hunk 5]
```diff
@@ -15,8 +18,10 @@ public class CollectArgumentProcessorTests
     {
         private TestableRunSettingsProvider settingsProvider;
         private CollectArgumentExecutor executor;
+        private Architecture architecture;
+        private Framework frameWork;
         private const string DefaultRunSettings = "<?xml version=\"1.0\" encoding=\"utf-16\"?>\r\n<RunSettings>\r\n  <DataCollectionRunSettings>\r\n    <DataCollectors >{0}</DataCollectors>\r\n  </DataCollectionRunSettings>\r\n</RunSettings>";
-
+
```
Comment (harshjain2) [inline test/vstest.console.UnitTests/Processors/CollectArgumentProcessorTests.cs]: nit : remove whitespaces.

File: test/vstest.console.UnitTests/Processors/EnableCodeCoverageArgumentProcessorTests.cs [hunk 6]
```diff
@@ -6,12 +6,17 @@ namespace vstest.console.UnitTests.Processors
     using Microsoft.VisualStudio.TestPlatform.CommandLine.Processors;
     using Microsoft.VisualStudio.TestPlatform.Common;
     using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
```
Comment (harshjain2) [inline test/vstest.console.UnitTests/Processors/EnableCodeCoverageArgumentProcessorTests.cs]: Nit : put the using statements in correct sorted order.

File: src/vstest.console/Processors/CollectArgumentProcessor.cs [hunk 7]
```diff
@@ -195,5 +225,12 @@ private static void DisableUnConfiguredDataCollectors(DataCollectionRunSettings
 
             return false;
         }
+
+        private static void AppendChildNodeOrInnerText(XmlDocument doc, XmlElement owner, string elementName, string nameSpaceUri, string innerText)
```
Comment (harshjain2) [inline src/vstest.console/Processors/CollectArgumentProcessor.cs]: Consider re-using class `RunSettingsProviderExtensions`.

Comment (harshjain2) [inline src/vstest.console/Processors/CollectArgumentProcessor.cs]: https://github.com/Microsoft/vstest/blob/master/src/vstest.console/Processors/Utilities/RunSettingsProviderExtensions.cs#L89

File: src/vstest.console/Processors/CollectArgumentProcessor.cs [hunk 8]
```diff
@@ -7,6 +7,8 @@ namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors
     using System.Collections.Generic;
```
Comment (harshjain2) [inline src/vstest.console/Processors/CollectArgumentProcessor.cs]: Nit : Remove extra line.

File: src/Microsoft.TestPlatform.ObjectModel/DataCollector/DataCollectorSettings.cs [hunk 9]
```diff
@@ -227,6 +247,13 @@ internal static DataCollectorSettings FromXml(XmlReader reader)
             return settings;
         }
 
+        private static void AppendChildNodeOrInnerText(XmlDocument doc, XmlElement owner, string elementName, string nameSpaceUri, string innerText)
```
Comment (harshjain2) [inline src/Microsoft.TestPlatform.ObjectModel/DataCollector/DataCollectorSettings.cs]: Consider re-using class RunSettingsProviderExtensions.
 https://github.com/Microsoft/vstest/blob/master/src/vstest.console/Processors/Utilities/RunSettingsProviderExtensions.cs#L89

**Review by harshjain2 — CHANGES_REQUESTED** [review]
Few concerns : 
1. This PR involves changes in Object Model, which I believe is not required. 
2. xlf files are not included for all languages. 

Suggestions : 
1. If we are planing to pass complete runsettings to data collectors in future, then just scope these changes only in datacollector process.
2. Can we make the required changes to pass the runsettings to data collectors now instead of having an intermediate state?

**Comment by mayankbansal018** [comment]
It was discussed, that this will be an intermediate state to enable Code Coverage for net core.
The idea to pass complete runsettings to datacollectors is still not completely closed.

**Pull request closed without merging**